src/Tests/StepDefinitions/CalculateEarningsForLearningPaymentsStepDefinitions.cs

```diff
@@ -1,60 +1,79 @@
-            .Create();
+        _messageHelper.CreateAnApprenticeshipMessage(start_date, planned_end_date, agreed_price);
+    }
 
-        await _context.Get<TestMessageBus>().Publish(_apprenticeshipCreatedEvent);
+    [When(@"the apprenticeship commitment is approved")]
+    public async Task TheApprenticeshipCommitmentIsApproved()
+    {
+        await _messageHelper.PublishAnApprenticeshipApprovedMessage();
+        await _messageHelper.ReadEarningsGeneratedMessage();
     }
 
-    [Then(@"Earnings results are published with calculated (.*), (.*), (.*), R(.*)-(.*), (.*)/(.*)")]
-    public async Task ThenEarningsResultsArePublishedWithCalculated(decimal adjustedAgreedPrice, decimal learningAmount, int numberOfInstalments, short firstDeliveryPeriod, short firstDeliveryAcademicYear, short firstCalendarPeriodMonth, short firstCalendarPeriodYear)
+    [Then(@"the total on-program payment amount must be calculated as 80% of the agreed price �(.*)")]
+    public void TheTotalOnProgramPaymentAmountMustBeCalculatedAs80PercentOfTheAgreedPrice(decimal adjustedPrice)
     {
-        await WaitHelper.WaitForIt(() => EarningsGeneratedEventHandler.ReceivedEvents.Any(), "Failed to find published event");
+        _context.Get<EarningsGeneratedEvent>().FundingPeriods.Should().HaveAdjustedAgreedPriceOf(adjustedPrice);
+    }
 
-        _earnings = EarningsGeneratedEventHandler.ReceivedEvents.First();
-        _earnings.FundingPeriods.Should().HaveAdjustedAgreedPriceOf(adjustedAgreedPrice);
+    [Then(@"the planned number of months must be the number of months from the start date to the planned end date (.*)")]
+    public void VerifyThePlannedDurationMonthsWithinTheEarningsGenerated(short numberOfInstallments)
+    {
+        _context.Get<FundingPeriod>().DeliveryPeriods.Should().HaveCount(numberOfInstallments);
+    }
 
-        var firstFundingPeriod = _earnings.FundingPeriods.First();
-        firstFundingPeriod.DeliveryPeriods.Should().HaveCount(numberOfInstalments);
-        firstFundingPeriod.AgreedPrice.Should().Be(adjustedAgreedPrice);
-        firstFundingPeriod.DeliveryPeriods.ForEach(dp => dp.LearningAmount.Should().Be(learningAmount));
+    [Then(@"the instalment amount must be calculated by dividing the total on-program amount equally into the number of planned months (.*)")]
+    public void VerifyInstallmentAmountIsCalculatedEquallyIntoAllEarningMonths(decimal installmentAmount)
+    {
+        _context.Get<FundingPeriod>().DeliveryPeriods.ForEach(dp => dp.LearningAmount.Should().Be(installmentAmount));
+    }
 
-        firstFundingPeriod.DeliveryPeriods.ShouldHaveCorrectFundingPeriods(numberOfInstalments, firstDeliveryPeriod, firstDeliveryAcademicYear);
-        firstFundingPeriod.DeliveryPeriods.ShouldHaveCorrectFundingCalendarMonths(numberOfInstalments, firstCalendarPeriodMonth, firstCalendarPeriodYear);
+    [Then(@"Earnings generated for each month starting from the first delivery period R(.*)-(.*) and first calendar period (.*)/(.*)")]
```

Comment:
Do delivery periods start with R? (Collection periods do, but I thought delivery periods did not)